lib/id3c/cli/command/etl/presence_absence.py

```diff
@@ -285,13 +290,27 @@ def sample_details(document: dict) -> dict:
         },
     }
 
-def presence_absence_details(document: dict, chip: Any = None) -> dict:
+def presence_absence_details(document: dict,
+                             chip: Any = None,
+                             assay_name: Any = None,
+                             assay_date: Any = None) -> dict:
     """
     Describe presence/absence details in a simple data structure designed to
     be used from SQL.
+
+    Raises `AssertionError` if we find an unknown *assay_name*.
     """
+    device = None
+
+    if assay_name:
+        assert assay_name in {'OpenArray', 'TaqmanQPCR'}, f"Found unknown assay name «{assay_name}»"
+        device = assay_name
+    elif chip:
+        device = "OpenArray"
+
     return {
-        "device": "OpenArray" if chip else None,
+        "device": device,
+        "assay_date": assay_date,
```

Comment:
Do we want `null` assay_date for p/a records where that isn't available?

Reply:
That's what we've done with `device` so I followed the same pattern. 
My thought is this way we know we've tried to ingest `assay_date` but the value is `null` because it was not provided to us.